text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+- Feature Name: Extended Standard Library
+- Start Date: 2025-05-08
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC describes an Extended Standard Library (ESL) for Rust. The ESL will consist of a set of crates providing commonly used functionality, subject to policies which promote excellent security, quality, and ease of use.
+
+# Motivation
+[motivation]: #motivation
+
+Rust programs, on average, have many dependencies. This is largely because the Rust standard library is intentionally limited in scope. As a consequence, Rust programs are currently required to pull in many third-party dependencies to get functionality that is considered table-stakes by 2025 standards.
+
+There are three issues this proposal aims to help address, all of which are related to the large number of independent entities providing crates that are widely depended on in Rust software:
+
+### 1. Impractically-sized web of trust
+
+Reliance on a high number of third-party dependencies creates a large web of trust for any non-trivial program. Given the size of this web of trust, it is practically impossible to be confident about the trustworthiness of the full dependency tree.
+
+Being a _trustworthy_ dependency means it:
+
+* Is developed and maintained by qualified, non-malicious authors
```

Comment:
Non-malicious is subjective:  You can be upstanding and well intentioned, but your employer maybe considered inherently malicious, ala government contractors, blockchain companies, advertising companies.  It's likely that some nationalities get treated with suspicion by specific governments too, maybe requiring more ongoing auditing or whatever.  It's likely better if these problems get addressed behind closed doors by the worried parties, nd the wider community mostly ignores them.  

What might make more sense is if crates put "Integration notes" into their repos an/dor issues on what they've done, or plan to do, to worke better with other crates, or what they wish others would do.

Reply:
There needs to be a (more) objective criteria here. Otherwise, people, topics, or groups can simply be declared malicious, even though a dependency on the crates they develop is unlikely to intentionally cause security issues.

This is particularly important for a project that seeks to be the One True Source for the One True Crate for X. You don't want to get caught up in arguments about the value or purpose of crates. Those issues are important to discuss and act on, but they need to be out of scope for this process.

(There are already contentious examples in this RFC discussion.)